drivers/flash/flash_stm32h7x.c

```diff
@@ -836,17 +836,19 @@ static const struct flash_parameters *flash_stm32h7_get_parameters(const struct
 	return &flash_stm32h7_parameters;
 }
 
-#ifndef CONFIG_SOC_SERIES_STM32H7RSX
 /* Gives the total logical device size in bytes and return 0. */
 static int flash_stm32h7_get_size(const struct device *dev, uint64_t *size)
 {
 	ARG_UNUSED(dev);
 
+#ifdef CONFIG_SOC_SERIES_STM32H7RSX
+	*size = (uint64_t)0x10000U; /* The series has only 64K of user flash memory */
```

Comment:
I missed that: would be better with `KB(64)`, and you can drop the cast.
Not a strong opinion.